packages/cli/generation/ir-migrations/src/migrations/v12-to-v11/migrateFromV12ToV11.ts

```diff
@@ -25,95 +25,59 @@ export const V12_TO_V11_MIGRATION: IrMigration<
         [GeneratorName.POSTMAN]: AlwaysRunMigration,
     },
     migrateBackwards: (v12, { taskContext, targetGenerator }): IrVersions.V11.ir.IntermediateRepresentation => {
-        const v11Types: Record<string, IrVersions.V11.types.TypeDeclaration> = {};
-        Object.entries(v12.types).forEach(([typeName, typeDeclaration]) => {
-            IrVersions.V12.types.Type._visit(typeDeclaration.shape, {
-                union: (unionTypeDeclaration) => {
-                    v11Types[typeName] = {
-                        ...typeDeclaration,
-                        shape: IrVersions.V11.types.Type.union({
-                            ...unionTypeDeclaration,
-                        }),
-                    };
-                },
-                enum: (enumTypeDeclaration) => {
-                    v11Types[typeName] = {
-                        ...typeDeclaration,
-                        shape: IrVersions.V11.types.Type.enum({
-                            ...enumTypeDeclaration,
-                        }),
-                    };
-                },
-                object: (objectTypeDeclaration) => {
-                    v11Types[typeName] = {
-                        ...typeDeclaration,
-                        shape: IrVersions.V11.types.Type.object({
-                            ...objectTypeDeclaration,
-                        }),
-                    };
-                },
-                alias: (aliasTypeDeclaration) => {
-                    IrVersions.V12.types.ResolvedTypeReference._visit(aliasTypeDeclaration.resolvedType, {
-                        container: (containerType) => {
-                            v11Types[typeName] = {
-                                ...typeDeclaration,
-                                shape: IrVersions.V11.types.Type.alias({
-                                    resolvedType: IrVersions.V11.types.ResolvedTypeReference.container(containerType),
-                                    aliasOf: aliasTypeDeclaration.aliasOf,
-                                }),
-                            };
+        const v11Types: Record<string, IrVersions.V11.types.TypeDeclaration> = mapValues(
+            v12.types,
+            (typeDeclaration) => {
+                return {
+                    ...typeDeclaration,
+                    shape: IrVersions.V12.types.Type._visit<IrVersions.V11.types.Type>(typeDeclaration.shape, {
+                        union: IrVersions.V11.types.Type.union,
+                        enum: IrVersions.V11.types.Type.enum,
+                        object: IrVersions.V11.types.Type.object,
```

Comment:
So this is pointing at the constructor?

Reply:
yeah exactly

`IrVersions.V11.types.Type.object` is a reference to a function, so it's equivalent to

```ts
object: (...args) => IrVersions.V11.types.Type.object(...args)
```